meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -247,4 +249,39 @@ public boolean[] set(List keys, List values, List version) {
   public List<OpResult> transactionOP(Iterable iterable) {
     return null;
   }
+
+  /**
+   * A converter class to transform {@link DataChangeListener} to {@link IZkDataListener}
+   */
+  static class DataListenerConverter implements IZkDataListener {
```

Comment:
May I ask why we need this class? Can zkMetaCLient implement IZkDataListener?

Reply:
This converter will be eventually added as IZkDataListener to zkclient. I don't think we want to make zkMetaCLient itself an listener.